src/core/radio/trel_link.cpp

```diff
@@ -373,8 +376,42 @@ void Link::ProcessReceivedPacket(Packet &aPacket)
 
     Get<Mac::Mac>().HandleReceivedFrame(&mRxFrame, kErrorNone);
 
+    // As the received frame is processed by the MAC or MLE layers,
+    // `CheckPeerAddrOnRxSuccess()` may be called with different modes,
+    // depending on whether the frame passes receive security checks
+    // at either the MAC or MLE layers, allowing or disallowing peer
+    // socket address to be updated from received TREL packet info.
+
 exit:
-    return;
+    mRxPacketPeer = nullptr;
+}
+
+void Link::CheckPeerAddrOnRxSuccess(PeerSockAddrUpdateMode aMode)
+{
+    Ip6::SockAddr prevSockAddr;
+
+    VerifyOrExit(mState != kStateDisabled);
+
+    VerifyOrExit(mRxPacketPeer != nullptr);
+
+    prevSockAddr = mRxPacketPeer->GetSockAddr();
+    VerifyOrExit(prevSockAddr != mRxPacketSenderAddr);
+
+    LogNote("Peer %s rx sock-addr differs from previously discovered by platform",
```

Comment:
the message is, in general, not always correct. If a peer-address has been already updated once before, then it's not necessarily different from the discovered address(es).  It's just different from the previously used peer address.

Reply:
Updated the log to say "from previously saved" instead.